### PR TITLE
adding a SlackFunctionTester to help with writing function tests

### PR DESCRIPTION
**Pull request by selfcontained**
## Summary
This creates a new top level export of `SlackFunctionTester` that can be used to ease writing tests for Slack Functions. Consuming it would like like the following when writing a test for  a function in your project:

```ts
// Can call this anywhere in your test file
const { createContext } = SlackFunctionTester("reverse_string");
```

And then in your tests where you need to pass arguments into your function...
```ts
const { outputs } = await ReverseString(createContext({ inputs }));


A full test would like like the following:
```ts
import { assertEquals } from "https://deno.land/std@0.99.0/testing/asserts.ts";
import ReverseString from "./reverse.ts";
import { SlackFunctionTester } from "deno-slack-sdk/mod.ts";

const { createContext } = SlackFunctionTester("reverse_string");

Deno.test("Reverse string function test", async () => {
  const inputs = { stringToReverse: "foo" };
  const { outputs } = await ReverseString(createContext({ inputs }));
  assertEquals(outputs?.reverseString, "oof");
});
```

I think we could expand on what helper functions `SlackFunctionTester()` returns if we need over time, but initially this will help developers have more realistic arguments into their functions w/ low effort.

### CODE REVIEW

**Comment by selfcontained** [comment]
Assuming we are ok with this approach, and after any changes, I can go ahead and get the default project template test updated to use this.

File: src/functions/function_tester.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { Env } from "../types.ts";
+import { FunctionContext, FunctionInvocationBody } from "./types.ts";
+
+type SlackFunctionTesterArgs<InputParameters> = {
+  inputs: InputParameters;
+  env?: Env;
+  token?: string;
+  event?: FunctionInvocationBody["event"];
+};
```
Comment (shapirone) [inline src/functions/function_tester.ts:9]: nit: I'm thinking that we should tie this a bit closer to the `FunctionContext` itself and we could leverage a `Partial` to make things optional. I'm thinking something like

```suggestion
import type { FunctionContext } from "./types.ts";

type SlackFunctionTesterArgs<InputParameters> =
  & Partial<FunctionContext<InputParameters>>
  & {
    inputs: InputParameters;
  };
```

**Review by shapirone — APPROVED** [review]
This LGTM and sets up a nice precedent for test helpers if we ever need to provide more 🙌 

Just one optional nit suggestion!

File: src/functions/function_tester.ts (hunk 1 above)
Comment (selfcontained) [reply src/functions/function_tester.ts:9]: Good call, helps it stay in sync w/ that type.

**Review by shapirone — APPROVED** [review]
:chefs-kiss:

**Pull request merged**